### PR TITLE
Chplvis

### PR DESCRIPTION
**Pull request by tomyhoi**
Fix regressions from 1.12.0 to HEAD.
Update version to 1.2 for data files.
Dump file name table, use file numbers in data lines.
Dump tag names on locale 0, use tag numbers in data lines.

### CODE REVIEW

**Comment by tomyhoi** [comment]
This PR contains some debug code removal and minor formatting changes for consistency.

File: modules/packages/VisualDebug.chpl [hunk 1]
```diff
@@ -130,50 +132,28 @@ private proc VDebugTree (what: vis_op, name: string, time: real, id: int = 0,
   :arg tagname: name of the tag
 */
   proc tagVdebug ( tagname : string ) {
-    //coforall l in Locales[1..] do
-    //  on l do chpl_vdebug_tag (tagname.localize().c_str(), 0);
-    //chpl_vdebug_tag (tagname.localize().c_str(), 0);
     if (VisualDebugOn) {
-       VDebugTree (vis_op.v_tag, tagname, 0);
+       chpl_vdebug_tagname (tagname.c_str(), tagno);
```
Comment (gbtitus) [inline modules/packages/VisualDebug.chpl]: This seems really parallel-unsafe (global var updated without any protection). Don't we need something here to serialize execution when more than one task arrives here at the same time?  Or is there something else that prevents that from happening?

**Comment by gbtitus** [comment]
Looks good modulo the line comment.

File: modules/packages/VisualDebug.chpl (hunk 1 above)
Comment (tomyhoi) [inline modules/packages/VisualDebug.chpl]: That is a good question.   tagVdebug() is supposed to be called only on locale-0.    Having two different tagVdebug() calls happening at the same time from two or more tasks would also cause problems for chplvis in trying to figure out the data.   I would have no problem making sure this code runs uninterrupted.

**Comment by tomyhoi** [comment]
@ronawho, since Greg is gone, would you check this out out.

File: modules/packages/VisualDebug.chpl [hunk 2]
```diff
@@ -122,6 +122,7 @@ private proc VDebugTree (what: vis_op, name: string, time: real, tagno: int,
   proc startVdebug ( rootname : string ) {
     var now = chpl_now_time();
     if (VisualDebugOn) {
+      tagno.write(0);
```
Comment (ronawho) [inline modules/packages/VisualDebug.chpl]: tagno will default to 0, so you don't have to explicitly initialize it

**Comment by ronawho** [comment]
I didn't review super carefully, but looks fine to me.

**Pull request merged**